packages/requisitions/src/RequestRequisition/api/hooks.ts

```diff
@@ -202,3 +204,56 @@ export const useDeleteSelectedRequisitions = () => {
 
   return deleteAction;
 };
+
+export const useAddFromMasterList = () => {
+  const { error } = useNotification();
+  const queryClient = useQueryClient();
+  const { id: requestId, requisitionNumber } = useRequestFields([
+    'id',
+    'requisitionNumber',
+  ]);
+  const api = useRequestApi();
+  const mutationState = useMutation(api.addFromMasterList, {
+    onSettled: () =>
+      queryClient.invalidateQueries(api.keys.detail(String(requisitionNumber))),
+  });
+
+  const t = useTranslation('distribution');
+  const getConfirmation = useConfirmationModal({
+    title: t('heading.are-you-sure'),
+    message: t('message.confirm-add-from-master-list'),
+  });
+
+  const addFromMasterList = async ({
+    id: masterListId,
+  }: MasterListRowFragment) => {
+    getConfirmation({
+      onConfirm: () =>
+        mutationState.mutate(
+          { masterListId, requestId },
+          {
+            onError: e => {
+              const { message } = e as Error;
+              switch (message) {
+                case 'CannotEditRequisition': {
+                  return error('Cannot edit requisition')();
+                }
+                case 'RecordNotFound': {
+                  return error('This master list has been deleted!')();
+                }
+                case 'MasterListNotFoundForThisStore': {
+                  return error(
+                    "Uh oh this is not the master list you're looking for"
+                  )();
+                }
+                default:
+                  return error('Could not add items to requisition')();
+              }
```

Comment:
I didn't spend time adding translations for this yet. I am not sure if it'll stay?

I *think* it's a decent idea.. essentially throw the error 'code' from the api fn and then switch on it.. likely a better implementation and maybe worth some iterations.